### PR TITLE
fix: the original error is shown when retrieving csrf token

### PR DESCRIPTION
**Pull request by jjtang1985**
## Context

In e2e test vdm, batch failed with OP destination, the error was:
`The error response is undefined.\n    at /home/vcap/app/node_modules/@sap-cloud-sdk/core/src/header-builder/csrf-token-header.ts`

## Definition of Done

Please consider all items and remove only if not applicable.

- [ ] Tests created/adjusted for your changes.
- [x] Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [x] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
- [ ] If applicable: Properly documented (JSDoc of public API)
- [ ] If applicable: Check if `node run doc` still works.

### CODE REVIEW

File: packages/core/src/header-builder/csrf-token-header.ts [hunk 1]
```diff
@@ -49,7 +49,9 @@ function makeCsrfRequest<T extends HttpRequestConfig>(
     .then(response => response.headers)
     .catch(error => {
       if (!error.response) {
-        throw new Error('The error response is undefined.');
+        throw new Error(
```
Comment (FrankEssenberger) [inline packages/core/src/header-builder/csrf-token-header.ts:52]: throw errorWithCause('The message'.error)

**Review by FrankEssenberger — APPROVED** [review]
LGTM align with marika if we should adjust the `errorWithCause` method to also include the `message` property of the exception goin in if it is present.

**Comment by jjtang1985** [comment]
Discussed with Marika, the 'stack' is not standard, but supported by all the browsers and node versions, which is reliable.

**Comment by FrankEssenberger** [comment]
What about her thoughts on the `message` property?

**Comment by jjtang1985** [comment]
> What about her thoughts on the message property?
Usually, the useful information is included in the stack so message is relatively useless. So we decided to use the existing `errorWithCause` unless we see some issues from the stack.

**Pull request merged**